### ISSUE TITLE
[JENKINS-68711] Script approval page does not display new scripts.

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#18097]
<p>Hello team,<br/>
We are experiencing new issue with jenkins, since version 2.332.3.<br/>
First, after update to this version it became impossible to approve scripts from the job page, which has this script.<br/>
After that we started to use page /scriptApproval/ to approve all scripts in jenkins.</p>

<p>And since yesterday (07.06.2022) new scripts stopped to appear on this page, but when we open job with "Configure" button, it still displays under script that it's not approved.</p>

<p>Luckily we have investigated, that by applying this configuration again (saving job without any changes by admin) - will lead script to appear in /scriptApproval/ page.</p>

<p>No changes were applied to this jenkins so far, last restart and maintenance was on may, 20.</p>

<p>Just for information, we are uploading our jobs with jenkins job builder script.</p>

<p>Also, running script </p>

<pre>
<span class="code-keyword">import</span> org.jenkinsci.plugins.scriptsecurity.scripts.*
toApprove = ScriptApproval.get().getPendingScripts().collect()
toApprove.each {pending -> ScriptApproval.get().approveScript(pending.getHash())} </pre>
<p>return nothing, like there is no not approved scripts in system.</p>

<p><b>Added after invesitgations:</b><br/>
Issue can be reproduced by clicking "Clear Approvals" button for scripts(remove $JENKINS_HOME/scriptApproval.xml works same). This will cause Jenkins to forget all approved scripts and they will be blocked on launch.<br/>
Moreover, script won't appear in script approval page until next restart. And on restart, it takes time to collect all scripts for that page blocking jenkins restart. We have more than 4000 jobs in our jenkins and it takes us roughly 1 hour to restart jenkins after approval clear.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=borys_tsolin">borys_tsolin</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-68711" target="_blank">Script approval page does not display new scripts.</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core, script-security-plugin
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 1
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Hello team,
We are experiencing new issue with jenkins, since version 2.332.3.
First, after update to this version it became impossible to approve scripts from the job page, which has this script.
After that we started to use page /scriptApproval/ to approve all scripts in jenkins.</p>

<p>And since yesterday (07.06.2022) new scripts stopped to appear on this page, but when we open job with "Configure" button, it still displays under script that it's not approved.</p>

<p>Luckily we have investigated, that by applying this configuration again (saving job without any changes by admin) - will lead script to appear in /scriptApproval/ page.</p>

<p>No changes were applied to this jenkins so far, last restart and maintenance was on may, 20.</p>

<p>Just for information, we are uploading our jobs with jenkins job builder script.</p>

<p>Also, running script </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">import</span> org.jenkinsci.plugins.scriptsecurity.scripts.*
toApprove = ScriptApproval.get().getPendingScripts().collect()
toApprove.each {pending -&gt; ScriptApproval.get().approveScript(pending.getHash())} </pre>
</div></div>
<p>return nothing, like there is no not approved scripts in system.</p>

<p><b>Added after invesitgations:</b>
Issue can be reproduced by clicking "Clear Approvals" button for scripts(remove $JENKINS_HOME/scriptApproval.xml works same). This will cause Jenkins to forget all approved scripts and they will be blocked on launch.
Moreover, script won't appear in script approval page until next restart. And on restart, it takes time to collect all scripts for that page blocking jenkins restart. We have more than 4000 jobs in our jenkins and it takes us roughly 1 hour to restart jenkins after approval clear.</p></pre>
</details>
</details>
